src/components/Tabs/utilities.ts

```diff
@@ -0,0 +1,41 @@
+import {TabDescriptor} from './Tabs';
```

Comment:
This causes a circular dependency which we want to avoid as results in confusing dependency graphs and makes the build do more work than it needs.

Extracting TabDescriptor out into a types.ts and having both utilities.ts and Tabs.ts both depend upon types.ts feels unnecessaryly. It think this function was fine where it originally was in Tabs.ts.

I think it's only worth extracting out utility functions when they're used in multiple locations. Otherwise keep the function in the only file that it is used in.

Reply:
@BPScott I think in this case it makes sense to extract the utility method to not overcomplicate the component tests and instead be able to just mock `getVisibleAndHiddenTabIndices` and check if its being called. `getVisibleAndHiddenTabIndices` can then have its own unit tests to make sure its actually reliable. More about this here: https://github.com/Shopify/polaris-react/pull/466/files#r228556790.